lib/Sema/TypeCheckDecl.cpp

```diff
@@ -975,6 +976,19 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
           isAcceptableVersionBasedChange = true;
         }
       }
+      // - or they are computed properties of different types,
```

Comment:
That's fine as long as the one that's available is the one in the current language mode.